math/convertall/Portfile

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           python 1.0
+PortGroup           qt5 1.0
+
+github.setup        doug-101 ConvertAll 0.8.0 v
+github.tarball_from releases
+
+python.versions     34 35 36 37 38 39
+python.default_version     38
+
+qt5.min_version     5.4
+
+name                convertall
+version             0.8.0
+license             GPL-2+
+categories          math science
+platforms           darwin
+maintainers         {@khorton kilohotel.com:kevin01}
+description         Extremely flexible unit converter
+long_description    ConvertAll has a large database of units, and allows conversions \
+                    that use multiple units, e.g. convert from feet per decade to \
+                    nautical miles per fortnight.
+homepage            http://convertall.bellz.org
+
+checksums           rmd160  89fcc2aa9bad7ecc1de7bee4edea68fdff39706a \
+                    sha256  624c8a792b0bc7ff3776499c2c743b32273569efd0567615e570a7e739e8d521 \
+                    size    281055
+
+depends_lib-append  port:py${python.version}-pyqt5
+
+use_configure       no
+build.cmd           ${python.bin} install.py
+build.args          -p ${prefix} \
+                    -b ${destroot}
+
+destroot {}
```

Comment:
this is kind of odd. No destroot phase? How does it install anything ?

Reply:
oh, perhaps it installs during the build.cmd phase? I see it runs install.py there.

Reply:
I don't see an easy way to split the install.py actions into build and destroot phases, so I propose to leave the portfile as-is

Reply:
I'm willing to agree with @kencu here. I'm not sure how yet, but we definitely need a destroot stage here. Otherwise things start to breakdown. 

As an example, I'm not sure how you would install buildbot binaries of this port, and surely the destroot dir wouldn't even exist during the build stage?

Maybe it's something to raise upstream, since they have a very unconventional install method. Alternatively, we could try to bypass the script entirely and write our own build stage based on the script.

Reply:
Would it be OK to move the install.py actions into the destroot phase, and declare an empty build phase?  

The docs show an example of an empty build phase in [section 4.3.4](https://guide.macports.org/#development.examples.eliminate), so hopefully this doesn't break the buildbot.

ConvertAll seems to not be actively updated now.  The last commit was over a year ago.  His build system apparently works correctly on Linux, so I doubt he would be interested in revising it to fit MacPorts requirements. 

Writing a custom build phase is a bit more than I was hoping to tackle for my first Portfile

Reply:
As long as the files to be installed end up in the destroot directory by the time that the destroot phase is over, that's fine. Typically a build would happen in the build phase and the built files would be copied into the destroot directory in the destroot phase. But if this build system does not easily accommodate splitting those tasks it is acceptable for both tasks to occur in a single phase (either build or destroot) and for the other one to be made a no-op. The [mongodb Portfile](https://github.com/macports/macports-ports/blob/d4bd2a5edc85e9aeb307eb7f8e9d58a5a2f582fc/databases/mongodb/Portfile#L136-L138) is an example of a port that already does that. My feeling today is that in this case having both build and destroot tasks take place in the build phase is preferable, since users usually expect the build phase to take longer than the destroot phase. (MacPorts also adds a `-j` flag to enable parallel building in the build phase but not in the destroot phase, though that is not relevant to this port since its build command is not `make`.) In the mongodb port, the build installs some things to the wrong places within the destroot directory, so I use a very brief destroot phase to clean that up.